cobald_hep_plugins/stopper.py

```diff
@@ -0,0 +1,65 @@
+from cobald.interfaces import Pool, PoolDecorator
+
+from cobald.utility import enforce
+
+import asyncio
+
+from cobald.daemon import service
+
+
+@service(flavour=asyncio)
+class Stopper(PoolDecorator):
+    """
+    Decorator that sets demand to 0 if the partition has no pending jobs
+
+    :param target: the pool
+    :param script: script that checks for pending jobs on the partition
+    :param interval: interval in seconds between execution of the script
+
+    If there are pending jobs on the partition, the demand is not modified.
+    The demand is set to 0 as long as no pending jobs are detected.
+
+    The default interval is 300 (5 minutes). The script has to be specified.
+    """
+
+    @property
+    def demand(self) -> float:
+        return self._demand
```

Comment:
Since the demand is just passed along unchanged, there is no need to store it on the class. You can fetch it from the target.
```suggestion
    @property
    def demand(self) -> float:
        return self.target.demand
```